app/(tabs)/notes.tsx

```diff
@@ -0,0 +1,86 @@
+import { Poppins_400Regular, useFonts } from '@expo-google-fonts/poppins';
+import { Redirect, useFocusEffect } from 'expo-router';
+import { useCallback, useState } from 'react';
+import { FlatList, SafeAreaView, StyleSheet } from 'react-native';
+import NoteItem from '../../components/NotesItem';
+import { colors } from '../../constants/colors';
+import type { Note } from '../../migrations/00003-createTableNotes';
+import type { NotesResponseBodyGet } from '../api/notes+api';
+import type { UserResponseBodyGet } from '../api/user+api';
+
+const styles = StyleSheet.create({
+  container: {
+    backgroundColor: colors.background,
+    height: '100%',
+  },
+  text: {
+    color: colors.text,
+  },
+  list: {
+    marginTop: 30,
+  },
+});
+
+export default function App() {
+  const [notes, setNotes] = useState<Note[]>([]);
+  const [fontsLoaded] = useFonts({
+    Poppins_400Regular,
+  });
+  const [isStale, setIsStale] = useState(true);
+
+  const renderItem = (item: { item: Note }) => <NoteItem note={item.item} />;
+
+  useFocusEffect(
+    useCallback(() => {
+      if (!isStale) return;
+
+      async function getUser() {
+        const response = await fetch('/api/user');
+
+        const body: UserResponseBodyGet = await response.json();
+
+        if ('error' in body) {
+          return <Redirect href="/(auth)" />;
+        }
+      }
+
+      async function getNotes() {
+        const response = await fetch('/api/notes');
+        const body: NotesResponseBodyGet = await response.json();
+
+        if ('error' in body) {
+          setNotes([]);
+        }
+
+        if ('notes' in body) {
+          setNotes(body.notes);
+        }
+
+        setIsStale(false);
+      }
+
+      getUser().catch((error) => {
+        console.error(error);
+      });
+
+      getNotes().catch((error) => {
+        console.error(error);
+      });
```

Comment:
Why are these 2 separate functions? If you're trying to start the `fetch` calls at the same time, you can also do this inside 1 function using `Promise.all()`

Reply:
8f58c4a5199ab84da2c0c8ad5788b98c8f893b4f

Reply:
I mean that there should only be something like 1 single `getUserAndNotes`, and inside of that, a Promise.all() call similar to this:

```ts
Promise.all([
  fetch(...),
  fetch(...),
```

Reply:
this was added

- https://github.com/upleveled/expo-example-fall-2024-atvie/pull/1/files#diff-b66174d7e773c79ec0463a26ffe42bdc0135ae76fb08792b4174f3a519632a04R40-R47